### PR TITLE
feat(ecam): Add ap&athr warning

### LINKED ISSUE #3727: [BUG] Master Alarm for TakeOff with Parking Brake on cannot be switched off
**Mod Version**
  "built": "2021-02-25T21:43:01+00:00",
  "ref": "master",
  "sha": "472cf527726e63d24e95608b6ac66d3430ffaacb",
  "actor": "Benjozork",
  "event_name": "manual"

**Describe the bug**
The Master Caution Alert cannot be switched off, when on a Runway, setting Thrust to TakeOff and having the Parking Brake still set to on.

**To Reproduce**
1. Load a new flight on any runway at any airport
2. Set Thrust to TakeOff or Max and leave the Parking Brake on
3. Try to disable the alarm - it won't switch off, even if you set the parking brake to off and try again.

**Expected behavior**
Master alarm can be cleared and alarm sound disappears

**Actual behavior**
Master alarm remains on all the time

**References**
- none

**Additional context**

Was this working before/when did the issue start occurring?
Haven't tried this for a very long time, but it used to work fine.

Is this a problem in the vanilla unmodded game?
No, a master alarm does not appear in the unmodded game at all.

### PR DESCRIPTION
**Pull request by Kimbyeoungjang**
Fixes #3727 

## Summary of Changes

using #3281 warning system, add warning message when disconnect autopilot / autothrottle.
also, when disconnect autothrottle, caution alert will ringing.

## References

image
image
image

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by Kimbyeoungjang** [comment]
this using my warning logic. so some of my pr is merged(i wish), need to fix my other pr.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 1]
```diff
@@ -1145,6 +1145,22 @@ var A320_Neo_UpperECAM;
...
+                        }
+                    },
+                    {
+                        message: "AP OFF",
+                        style: "fail-3",
+                        important: true,
+                        isActive: () => {
+                            return this.getCachedSimVar("L:A32NX_AP_DISC", "Bool");
+                        }
+                    },
```
Comment (beheh) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1163]: The messages should be swapped, as AP OFF has a higher priority.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_Core.js [hunk 2]
```diff
@@ -12,7 +12,8 @@ class A32NX_Core {
             new A32NX_FWC(),
             new A32NX_GPWS(this),
             new A32NX_GroundReference(),
-            new A32NX_Speeds()
+            new A32NX_Speeds(),
+            new A32NX_Warning
```
Comment (beheh) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_Core.js:16]: There is no reason to introduce a new warning system, this logic should be handled by the Flight Warning Computer (A32NX_FWC). I suggest moving your code to A32NX_FWC and removing A32NX_Warning.

**Review by beheh — CHANGES_REQUESTED** [review]

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_Warning.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    constructor(_core) {
+        console.log('A32NX_Warning constructed');
+
+        this.warningLeft = false;
+        this.warningRight = false;
+        this.cautionLeft = false;
+        this.cautionRight = false;
+        this.AutothrottleWarningCanceled = false;
+        this.athrdeltaTime = 0;
+        this.apdeltaTime = 0;
```
Comment (beheh) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_Warning.js:11]: You can use logic nodes from A32NX_Logic for this. Look at the FWC for examples on how they work.

**Comment by Kimbyeoungjang** [comment]
working done but need to merge #3281 first.

**Comment by Kimbyeoungjang** [comment]
change to draft until merge my other pr.
also need to add flashing logic - current AP warning only turn on and off but real aircraft have a flashing master warning.

**Comment by Kimbyeoungjang** [comment]
reopen to other pr due to branch cleanup

**Pull request closed without merging**